macOS/LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Shared/Operations/DataBrokerProfileQueryOperationManager.swift

```diff
@@ -256,12 +215,150 @@ struct DataBrokerProfileQueryOperationManager: OperationsManager {
             throw error
         }
     }
-    // swiftlint:enable cyclomatic_complexity
 
-    private func sendProfileRemovedNotificationIfNecessary(userNotificationService: DataBrokerProtectionUserNotificationService, database: DataBrokerProtectionRepository) {
+    private func scheduleExtractedProfilesForOptOut(extractedProfiles: [ExtractedProfile],
+                                                    brokerProfileQueryData: BrokerProfileQueryData,
+                                                    brokerId: Int64,
+                                                    profileQueryId: Int64,
+                                                    database: DataBrokerProtectionRepository,
+                                                    eventPixels: DataBrokerProtectionEventPixels,
+                                                    stageCalculator: DataBrokerProtectionStageDurationCalculator) throws {
+        stageCalculator.fireScanSuccess(matchesFound: extractedProfiles.count)
+
+        let event = HistoryEvent(
+            brokerId: brokerId,
+            profileQueryId: profileQueryId,
+            type: .matchesFound(count: extractedProfiles.count)
+        )
+        try database.add(event)
```

Comment:
I'd be inclined to move this and the pixel firing above out of this function since it's an unrelated/unexpected side effect

Reply:
Yeah very good call, I think we'll end up doing that a lot as part of the TD too but this is a good change to make now ahead of time, lemme do that

Reply:
Okay, done - updated the comments as well